server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/GroupItemNode.java

```diff
@@ -14,39 +14,82 @@
 import lombok.ToString;
 import org.eclipse.lsp.cobol.core.messages.MessageTemplate;
 import org.eclipse.lsp.cobol.core.model.Locality;
 import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.tree.Node;
 import org.eclipse.lsp.cobol.core.model.tree.NodeType;
+import org.eclipse.lsp.cobol.core.model.variables.UsageFormat;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Stream;
 
 import static org.eclipse.lsp.cobol.core.model.tree.variables.VariableDefinitionUtil.EMPTY_STRUCTURE_MSG;
 
 /**
  * This value class represents a group item COBOL variable. Group elements can have nested
  * variables. They cannot have neither PIC nor VALUE clauses. Can be the top element of the
  * structure with level 01, as well as nested ones with higher level numbers.
+ *
+ * <p>Group item can have usage clause.<br>
+ * <br>
+ *
+ * <pre>
+ *  Case 1.
+ *    Working-Storage Section.
+ *    01 FOO POINTER GLOBAL.   <-- This is valid. Here FOO node has
+ *    01 FOO2 pointer GLOBAL.       only 1 children of node type
+ *    Procedure Division.           VARIABLE_DEFINITION_NAME. Which should not
+ *                                  be considered for empty structure.
+ *     </pre>
+ *
+ * <pre>
+ *    Case 2.
+ *       WORKING-STORAGE SECTION.
+ *       01 PARENT.    <--- This is invalid
+ *       PROCEDURE DIVISION.
+ * </pre>
  */
 @Getter
 @ToString(callSuper = true)
-public class GroupItemNode extends VariableWithLevelNode {
+@EqualsAndHashCode(callSuper = true)
+public class GroupItemNode extends VariableWithLevelNode implements UsageClause {
   private boolean redefines;
+  private UsageFormat usageFormat;
 
-  public GroupItemNode(Locality location, int level, String name, boolean global, boolean redefines) {
+  public GroupItemNode(
+      Locality location,
+      int level,
+      String name,
+      boolean global,
+      boolean redefines,
+      UsageFormat usageFormat) {
     super(location, level, name, VariableType.GROUP_ITEM, global);
     this.redefines = redefines;
+    this.usageFormat = usageFormat;
   }
 
   @Override
   public List<SyntaxError> process() {
     List<SyntaxError> errors = new ArrayList<>();
-    if (getChildren().stream().noneMatch(node -> node.getNodeType() == NodeType.VARIABLE))
+    if (isValidChildrenPresent())
       errors.add(getError(MessageTemplate.of(EMPTY_STRUCTURE_MSG, getName())));
     errors.addAll(super.process());
     return errors;
   }
+
+  private boolean isValidChildrenPresent() {
+    Stream<Node> nodeStream =
+        getChildren().stream()
+            .filter(node -> node.getNodeType() != NodeType.VARIABLE_DEFINITION_NAME);
+    if (isRedefines())
+      nodeStream = nodeStream.filter(node -> node.getNodeType() != NodeType.VARIABLE_USAGE);
+
+    return (nodeStream.count() != 0 || getUsageFormat() == UsageFormat.UNDEFINED)
+        && getChildren().stream().noneMatch(node -> node.getNodeType() == NodeType.VARIABLE);
```

Comment:
Please, rewrite it to use `VARIABLE` nodes only.
As you see you already have two exceptions here for `VARIABLE_DEFINITION_NAME` and `VARIABLE_USAGE`. What if in the future we introduce a new node for some meta information?
But maybe, you don't need the method at all.
Anyway, the `Node` class can generate a neat predicate for filtering:
```
getChildren().stream().noneMatch(hasType(NodeType.VARIABLE))
```

Reply:
I understand the concern. But I am out of idea here. I Will work on it!

Reply:
I mean that you can count only `NodeType.VARIABLE`, you don't need to filter the other types. And the result can be like:
```
return variables.count() != 0 ^ getUsageFormat() != UsageFormat.UNDEFINED;
```

Reply:
😄 Why didn't this clicked to me? I will make that change. Thanks

Reply:
Done

Reply:
Cool! It's better than I thought initially.